### PR TITLE
feat(query): add bool_and and bool_or agg functions

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

feat(query): add bool_and and bool_or agg functions
 

Function | Argument Type(s) | Return Type | Description
-- | -- | -- | --
bool_and(expression) | bool | bool | true if all input values are true, otherwise false
bool_or(expression) | bool | bool | true if at least one input value is true, otherwise false

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/databendlabs/databend/18132)

### CODE REVIEW

File: tests/sqllogictests/suites/query/functions/02_0000_function_aggregate_bool.test [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+query B
+select bool_or(t) from (values (true), (true), (false)) a(t);
+----
+1
+
+query B
+select bool_or(t) from (values (false), (false), (false)) a(t);
+----
+0
```
Comment (KKould) [inline tests/sqllogictests/suites/query/functions/02_0000_function_aggregate_bool.test:29]: I think we can add `select bool_or(t) from (values ​​(null)) a(t);` to test whether empty data returns null
